db/schema.rb

```diff
@@ -1081,6 +1081,13 @@
     t.index ["user_id"], name: "index_sponsorships_on_user_id"
   end
 
+  create_table "suspended_users", force: :cascade do |t|
```

Comment:
[Question] I'm curious about the lack of a primary key on this table: could that impact searching for usernames (even though you have an index defined below)?

Reply:
tl;dr: While we could add an automatically incrementing integer primary key it would serve no purpose at all and effectively be a waste of storage space.

Longer version:

There's nothing really special about a primary key. It only adds non-null and unique constraints and is the column other tables use to reference this row. Primary keys can be surrogate (e.g. the `id` columns Rails adds by default), or natural (i.e. a regular column or combination of columns of the table that fulfills the above-listed constraints). 

We won't reference this model from other models, so the foreign key use case doesn't apply here. The requirements of uniqueness and presence are enforced through constraints. 

Primary keys don't factor into the query here, the only ones we perform against this table are fully covered by the index. In fact, even the index may be premature: unless a lot of suspended users start deleting their accounts, the query planner will most likely choose a sequential scan over an index scan anyway, as it generally does when the (a) the total number of rows is small enough or (b) a query would return the majority of the records in a table. 

With all that out of the way, I updated the migration to make the natural PK explicit rather than implicit. This is how the table looks now in Postgres:

```
PracticalDeveloper_development# \d suspended_users
                          Table "public.suspended_users"
┌───────────────┬────────────────────────────────┬───────────┬──────────┬─────────┐
│    Column     │              Type              │ Collation │ Nullable │ Default │
╞═══════════════╪════════════════════════════════╪═══════════╪══════════╪═════════╡
│ username_hash │ character varying              │           │ not null │         │
│ created_at    │ timestamp(6) without time zone │           │ not null │         │
│ updated_at    │ timestamp(6) without time zone │           │ not null │         │
└───────────────┴────────────────────────────────┴───────────┴──────────┴─────────┘
Indexes:
    "suspended_users_pkey" PRIMARY KEY, btree (username_hash)
```

Reply:
Cool, appreciate the deep dive @citizen428 and the index change!